### DISCUSSION TITLE
https://blog.marlam.in/how-to-human/

### DISCUSSION THREAD

**Discussion by giscus** [Marlamin/blog#4] [Announcements]
# https://blog.marlam.in/how-to-human/

Late night contemplating and reflecting is how I operate. It's when the world and by extension my brain is the most calm and actually able to collect thoughts. I don't know if this is normal or if there's something wrong up there, but that's how it is and has been as long as I can remember.

https://blog.marlam.in/how-to-human/

**Comment by xentec** [comment]
Hi, as someone with similar background and questions, I've been pondering about them for the last years and have arrived at some conjectures about 'how to make long term life decisions'. However, you'll need to salt and season them yourself. ;)

General advices:
- Speaking of salt and seasoning: ensure operation at optimal levels. If your body has shortcomings in nutrition, your mental speed and attention levels will deteriorate. Everyone has slightly different needs. Are you at your optimal level?
- If you want to look ahead more efficient, then learn more mathematics / rational thinking and apply it. This field is our oldest model / bet to date for predicting the future. 

Since you're uncomfortable with strangers (as was I), I suspect your empathy is underdeveloped and you also fear the unpredictable. The latter makes you more cautions and is more preferable for long term planning.
The former can be mitigated by learning the grammar of human interaction. You can do that either by reading themed books (e.g. "Surrounded by Idiots" by Thomas Erikson) or bite the bullet and interact with people outside (will be awkward and cringe at first, but you have a goal, remember?). You can also start easy by going to ComSci/hacker events (like MCH2022 or WHY2025) and meet alikes there. The key is to practice attention to the person and not drift away / to stay awake.
Once you've interacted with enough different personalities and start to see their common attributes, the fear/anxiety will gradually fade and in that new level of calmness you might be able to predict the behaviour of some people from a single look. Then, job interviews and teamwork won't be a problem any more!
Meanwhile, studying some math fields will enable you to see more interesting problems around you and abstract ways to solve them. Your ability as programmer will put the solution into practise. :)
Then you can build your career: on your own terms. [1]

[1]: https://youtu.be/pUCa31RGKRo

Hope these steps are a helpful way for you to leave your current local maximum. You may find others more seasoned for you. ;)
Are you ready to level up?

**Reply by Marlamin** [reply]
Thanks for your comment! I'll definitely keep some of these things in mind going forward.

**Comment by KalopsiaTwilight** [comment]
Aww, I wish I had read this earlier when it was posted. I did want to share a few words, even if I am a bit late to the party.

I can certainly relate to thinking a lot about the future. Looking back on it, I don't know if I ever got anywhere useful from contemplating. I think the problem with thinking about the future is asides from it being uncertain, is that our sense of self is also uncertain. I can look back at myself from 5 years ago and recognize that I am a very different person right now. And when I think about who I might be 5 years from now, I wouldn't even know who she is or what she might want. 

With that said, I do think there's a certain 'core' to my self that these versions of 'me' have in common. And that can be morals, hobbies, or more abstract ideas such as 'being creative'... Whatever it is, I think there are certain parts of our selves that we can recognize in all versions of us that makes up our 'core'. And I think that's where I go to for my 'important' decisions about the future. 

Typically, I tend to go with my gut. If something seems right, then it probably is. If it's something I have wanted to do but haven't for reasons... I probably should just do it! There's probably a good reason I want to go for it, even if I don't know exactly why. I actually think that a lot of people know intuitively what's good for them at the moment, they just have to find the strength to believe in their own answers and follow through.

That's usually easier said than done though. I know for myself there was a time where I was thinking about whether I should go further in my education or not. A lot of people in my environment thought it'd be a good idea. If you look at it on its face value, having a higher education tends to be pretty good for your future overall. It meant I wouldn't have to move out of my student housing and find another place to live... And there were no real negatives that weighed on the other side of the question. My gut feeling was that it wasn't right for me though. Despite that, I let myself be convinced by the people around me that it would be a good idea. I tried it out, and after a month decided to quit because I still didn't feel that it was right for me, I wasn't enjoying myself and I wanted to do other things. 

So I try to trust my gut more now and I've been very happy with where that has lead me! I think it's important to remind yourself that you don't need to have all the answers. You can just do what feels right to you and see where that leads you and don't feel too concerned about what a future you might think of you. I can look at my past self and see a lot of decisions that I would have made differently now, knowing what I know now. But I had to make those decisions to become the me that I am today. Every decision is an opportunity. Some might not work out but the ones that do, do! And I don't blame my past self for making the 'wrong' decision.

I say you have to be kind to all versions of yourself and let your 'core' or your gut guide you when you're making decisions about the future. That's what's lead to the decisions I've been happiest with.

**Reply by Marlamin** [reply]
No worries about being late, this kind of stuff is timeless anyways. I definitely recognize myself in being a very different person 5 years ago, but at the same time I still share many of the same things with him, most of which I've talked about in the passion blog post, so I definitely have some sort of sense of who I am at the core when it comes to those things but much less so when it comes to other yet undiscovered things that I know are there but that I just need to explore at some point to make part of said core. 

My intuition/gut has let me down in the past when exploring these which has led me to 'fall back' on things that are safe more often than not and pushing away things that are less safe. With slowly trying to adopt some of the things I mentioned in the original post, I'm hoping to build up enough courage to once again start exploring these things and maybe restore some of the faith in my gut/intuition to rely on it more.

I also wouldn't be where I am today without some of the decisions I made in the past. They might not have been the best decisions for the more distant future, but it's given me a lot of happiness, enjoyment and creative energy in the last few years so I can only thank my past self for making those specific decisions even though he specifically pushed off some things I'm still pushing off today.

Anyway, thanks for your reply, it was a great help. Also, thanks again for your work on WDBXEditor2, it is much appreciated.
